### PR TITLE
Issue/7368 update dialog style

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Update all dialog themes to use the same style as described in https://github.com/wordpress-mobile/WordPress-Android/issues/7368.  This is allow us to change the attributes of a [single style](https://github.com/wordpress-mobile/WordPress-Android/blob/36cfe4c6737308898da00433bc4acf748acfcfd9/WordPress/src/main/res/values/styles_calypso.xml#L56-L63) and have them reflected across the entire app for all dialogs [except login](https://github.com/wordpress-mobile/WordPress-Android/blob/36cfe4c6737308898da00433bc4acf748acfcfd9/libs/login/WordPressLoginFlow/src/main/res/values/styles.xml#L24-L31).

The only visible difference for most dialogs will be the button color change from [`orange_jazzy`](https://github.com/wordpress-mobile/WordPress-Android/blob/36cfe4c6737308898da00433bc4acf748acfcfd9/WordPress/src/main/res/values/colors.xml#L60) to [`blue_medium`](https://github.com/wordpress-mobile/WordPress-Android/blob/36cfe4c6737308898da00433bc4acf748acfcfd9/WordPress/src/main/res/values/colors.xml#L40).  See the screenshots below for illustration.

dialog_01

The biggest visible difference is with publish scheduling within post settings.  See the screenshots below for illustration.

dialog_02

dialog_03

### Test
The only way to test is to simply trigger dialogs throughout the app.  I separated each dialog type by commit so it would be easier to recognize which dialogs were changed.  The commit messages hint to which dialog type was changed.  Let me know if any are unclear.  I tried them all.

### CODE REVIEW

**Comment by oguzkocer** [comment]
@theck13 I just wanted to note that the reason we didn't use the date & time picker in edit post settings is that there was a rotation bug with the SDK: #4965. I couldn't test your branch to see if that issue still existed, because of #7463, but we should ensure the bug with the SDK is fixed before we merge this PR in.

**Comment by theck13** [comment]
The [`Calypso.Dialog` style parent was changed from `Theme.AppCompat.Light.Dialog.Alert` to `Theme.AppCompat.Light.Dialog`](https://github.com/wordpress-mobile/WordPress-Android/pull/7438/files#diff-f22a0ad9d36db03c6da756f5223f488eR56).  That eliminated the white space shown on the right side of the screenshots in https://github.com/wordpress-mobile/WordPress-Android/issues/4965.  Due to https://github.com/wordpress-mobile/WordPress-Android/issues/7463 like you mentioned, I can't test the layout on rotation change, but here are the dialogs in landscape orientation.

dialog_post_settings_landscape

**Comment by oguzkocer** [comment]
> Due to #7463 like you mentioned, I can't test the layout on rotation change, but here are the dialogs in landscape orientation.

It used to work in different orientations as well, it was just the rotation itself causing the issue. I am checking out #7463, but I don't think I'll be able to fix that today.

**Comment by theck13** [comment]
@oguzkocer, now that https://github.com/wordpress-mobile/WordPress-Android/pull/7541 and https://github.com/wordpress-mobile/WordPress-Android/pull/7542 are merged, I updated this branch and the dialogs are retained on device rotation.  Also, there are no odd layout issues as described in https://github.com/wordpress-mobile/WordPress-Android/issues/4965.

**Comment by oguzkocer** [comment]
Thanks for the update @theck13! That's great to hear 👍

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/RelatedPostsDialog.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
 
         toggleViews(mShowRelatedPosts.isChecked());
 
-        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity(), R.style.Calypso_AlertDialog);
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity(), R.style.Calypso_Dialog);
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/RelatedPostsDialog.java]: For the ones that used to rely on `Calypso_AlertDialog` I notice that we're not using `ContextThemeWrapper` - is that on purpose?

**Comment by nbradbury** [comment]
I'm not sure this is anything to worry about, or even if it's the result of this PR, but I noticed the dialog styles vary much more widely on API 16.

screenshot_1522090249

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/RelatedPostsDialog.java (hunk 1 above)
Comment (theck13) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/RelatedPostsDialog.java]: Nope, that was not intentional.  I must have missed those initially or during a merge conflict resolution.  Thanks for pointing it out!  I updated them in 5c1005b, ac551a0, 7ec2ab4, 770c6b4, 3a52c6c, 981c42e, 5b6dd26, ae7d2ca, and feca165.

**Comment by theck13** [comment]
> I'm not sure this is anything to worry about, or even if it's the result of this PR, but I noticed the dialog styles vary much more widely on API 16.

Those do look very different.  Since 1.7% of all Android users and 0.8% of WordPress users are on API 16, I would consider this negligible.

**Comment by nbradbury** [comment]
Looks good! :shipit:

**Pull request merged**